File: Aquecimento Prova/atleta.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct
{
    char nome[51];
    int peso
} Atleta;

void ordenar(Atleta *atletas[100], int n)
{
    for (int i = 0; i < n -1; i++){
        int maior = i;
    
        for (int j = i + 1; j < n; j++){
            if (atletas[maior]->peso < atletas[j]->peso || 
            (atletas[maior]->peso == atletas[j]->peso && strcmp(atletas[j]->nome, atletas[maior]->nome) < 0)){
                maior = j;
            }
        }
        if(maior != i){
            Atleta* tmp = atletas[i];
            atletas[i] = atletas[maior];
            atletas[maior] = tmp;
        }
    }
}

Atleta *newAtleta(char *nome, int peso)
{
    Atleta *atleta = malloc(sizeof(Atleta));
    strcpy(atleta->nome, nome);
    atleta->peso = peso;
    return atleta;
}
int main()
{
    Atleta *atletas[100];
    int i, n = 0;
    int peso;
    char nome[50];
    while (scanf(" %s %d", nome, &peso) != EOF)
    {
        atletas[i] = newAtleta(nome, peso);
        i++;
        n++;
    }

    ordenar(atletas, n);

    for(int i = 0; i<n; i++){
        printf(" %s %d\n", atletas[i]->nome, atletas[i] ->peso);
    }

    
    for(int i = 0; i<n; i++){
        free(atletas[i]);
    }
}

File: Aquecimento Prova/Atleta.java
import java.util.Scanner;

public class Atleta {
    public static class Atletas {
        String nome;
        int peso;
    }

    public static void ordenar(Atletas[] array) {
        for (int i = 0; i < array.length; i++) {
            int maior = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j].peso > array[maior].peso || (array[j].peso == array[maior].peso &&
                        array[j].nome.compareToIgnoreCase(array[maior].nome) < 0)) {
                    maior = j;
                }
            }
            Atletas tmp = array[i];
            array[i] = array[maior];
            array[maior] = tmp;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        Atletas[] array = new Atletas[n];
        
        // Preencher o array com n atletas
        for (int i = 0; i < n; i++) {
            array[i] = new Atletas();
            array[i].nome = sc.next();
            array[i].peso = sc.nextInt();
        }

        ordenar(array);

        // Imprimir os atletas ordenados
        for (int i = 0; i < n; i++) {
            System.out.println(array[i].nome + " " + array[i].peso);
        }
        
        sc.close();
    }
}